pkg/utils/flow/flow.go

```diff
@@ -213,7 +213,10 @@ func (e *execution) runNode(ctx context.Context, id TaskID) {
 		end := time.Now().UTC()
 		e.log.WithValues(logKeyTask, id).V(1).Info("Finished", "duration", end.Sub(start))
 
-		if err != nil {
+		if err == ErrTaskSkipped {
+			err = nil
+			e.log.WithValues(logKeyTask, id).Info("Skipped")
```

Comment:
```suggestion
			e.log.V(2).WithValues(logKeyTask, id).Info("Skipped")
```

WDYT?

Reply:
I have made the log level the same as https://github.com/gardener/gardener/blob/cd14f0a0859861f6a6afc7ab8a06621bc33d0da9/pkg/utils/flow/flow.go#L211-L214

Related to the discussion about shoot status. There `lastOperation.description` doesn't come as `skipped task -  name`, but only the `task name`, that should be fine. WDYT?

Reply:
It is not really possible not to add a task in the running, as there can be successive tasks dependent on skipped tasks.
For a task to get executed all the task on which it is dependent should succeed first - 

https://github.com/gardener/gardener/blob/b78492fc4f6777232f4744dc96f6f60b267396a6/pkg/utils/flow/flow.go#L237-L245

https://github.com/gardener/gardener/blob/b78492fc4f6777232f4744dc96f6f60b267396a6/pkg/utils/flow/flow.go#L283-L298

How about adding a skipped task list here - https://github.com/gardener/gardener/blob/b78492fc4f6777232f4744dc96f6f60b267396a6/pkg/utils/flow/flow.go#L122-L129

and if a task is in the skipped list, it will not be added here - https://github.com/gardener/gardener/blob/b78492fc4f6777232f4744dc96f6f60b267396a6/pkg/utils/flow/progress_reporter.go#L43

Reply:
Sure, sounds good @acumino